Convert readme to markdown

streeve opened the pull request:
This info is out of date, but at least converted from the previous doc. Moved MPI, Stork, etc. to a new PR

JamieStumpORNL reviewed (approved):
Good stuff

gknapp1 commented on README.md:
```diff
@@ -10,3 +17,155 @@ Heat transfer code utilizing a nondimensionalized semi-analytic solution to movi
+This section shows how to create a custom simulation with all tunable parameters.
+If a necessary keyword is not included, often the program will default to values found in the test case. Note when editing files, it is important to begin each group with a brace `{`, as well as close each group `}` as in the example files.
+
+When run from the command line, an input file is optional (e.g. `./3DThesis ParamInput.txt`), defaulting to `TestInputs/ParamInput.txt`. If `Name` is set to `TestSim`, then the data for the simulation can be found at `Data/TestSim`. All the files under `Simulation` are necessary, the rest are optional.
```
Did this change? I thought data was output to `TestSim/Data`.

gknapp1 commented on README.md:
```diff
@@ -10,3 +17,155 @@ Heat transfer code utilizing a nondimensionalized semi-analytic solution to movi
+  - `a`
+
+#### Beam File
+This file contains information on the energy source, which is a volumetric gaussian. It should be noted that everything input here is equivalent to `√6 σ`. For example, if the standard deviation of an actual beam is `20 µm` and radially symmetric, both `Width_X` and `Width_Y` should be set to 48.9898e-6.
```
Specify that $\sigma$ is as defined by the $D4\sigma$ beam diameter.

gknapp1 commented on README.md:
```diff
@@ -10,3 +17,155 @@ Heat transfer code utilizing a nondimensionalized semi-analytic solution to movi
+This section shows how to create a custom simulation with all tunable parameters.
+If a necessary keyword is not included, often the program will default to values found in the test case. Note when editing files, it is important to begin each group with a brace `{`, as well as close each group `}` as in the example files.
+
+When run from the command line, an input file is optional (e.g. `./3DThesis ParamInput.txt`), defaulting to `TestInputs/ParamInput.txt`. If `Name` is set to `TestSim`, then the data for the simulation can be found at `Data/TestSim`. All the files under `Simulation` are necessary, the rest are optional.
```
Did this change? I thought data was output to TestSim/Data.

JamieStumpORNL replied:
It changed TO TestSim/Data. Shows how old the readme is.

gknapp1 commented on README.md:
```diff
@@ -10,3 +17,155 @@ Heat transfer code utilizing a nondimensionalized semi-analytic solution to movi
+  - `Compression`: Binary toggle (0-off, 1-on) for using a path compression algorithm. This is primary useful for LARGE scan paths (1 Mb or larger) with smooth-ish movement of the heat source (ex: raster or point raster) rather than discontinuous movement (ex: random point fill). It does so by compressing multiple nearby diffuse heat sources into a single source. There is a small loss in accuracy ...
+- Compute
+  - `MaxThreads`: Number of threads to use for the simulation.
+  - `PINT`: Allow for Parallel-IN-Time calculations. NOTE: This has been temporarily removed during the code overhaul
```
Is PINT still on the roadmap for 3DThesis? Might want to remove this.

gknapp1 reviewed (changes requested):
A few things are missing in the markdown that are in the Word document:

- Getting started section
- Images for Visual Studio build instructions & for the sample output at the very end of the document.
- Note on usage of multiple heat source files

streeve commented:
> A few things are missing in the markdown that are in the Word document:
> 
>     - Getting started section
- Agreed

>     - Images for Visual Studio build instructions & for the sample output at the very end of the document.
- Not relevant & will add 

>     - Note on usage of multiple heat source files
- Not sure how this works 

Also
- MPI inputs
- Melt pool tracking

gknapp1 commented:
Is the Visual Studio build no longer being supported? Fine if so, but a few things to note: 

- It might help people who are already using 3DThesis to make a note saying to refer to the documentation included with the Version 3.0 release for details.
- There are notes on compiler options that are recommended. Are these all captured in the CMake build and would there be use cases where someone might need to know to change the CMake build options?

> To allow for parallel computing and vectorize calculations, some additional steps are needed (RECOMMENDED). All recommended changes can be found in the project properties (the big menu which contained the ability to add include directories). It is also recommended that the Platform is set to x64 and Configuration is changed to Active (unless debugging).
> 1.	Configuration Properties -> C/C++ -> Code Generation -> Floating Point Model = Fast
> 2.	Configuration Properties -> C/C++ -> Language -> Conformance mode = No
> 3.	Configuration Properties -> C/C++ -> Language -> Open MP Support = Yes

gknapp1 commented:
On the multiple beam usage, this is the note in the Word document. @JamieStumpORNL is this still a working feature?

> NOTE: To use multiple heat sources, use the * to denote the wildcard (ex: “Beam.*.txt”). The program will start at “Beam.1.txt” then “Beam.2.txt” and keep going until “Beam.x.txt” does not exist. The same goes for the paths. “Path.x.txt” would be used with the beam parameters found in “Beam.x.txt”

JamieStumpORNL commented:
> Is the Visual Studio build no longer being supported? Fine if so, but a few things to note:
> 
> * It might help people who are already using 3DThesis to make a note saying to refer to the documentation included with the Version 3.0 release for details.
> * There are notes on compiler options that are recommended. Are these all captured in the CMake build and would there be use cases where someone might need to know to change the CMake build options?
> 
> > To allow for parallel computing and vectorize calculations, some additional steps are needed (RECOMMENDED). All recommended changes can be found in the project properties (the big menu which contained the ability to add include directories). It is also recommended that the Platform is set to x64 and Configuration is changed to Active (unless debugging).
> > 
> > 1. Configuration Properties -> C/C++ -> Code Generation -> Floating Point Model = Fast
> > 2. Configuration Properties -> C/C++ -> Language -> Conformance mode = No
> > 3. Configuration Properties -> C/C++ -> Language -> Open MP Support = Yes

I guess not. It has been a while since I used VS but it should still build (before the mpi update)

> On the multiple beam usage, this is the note in the Word document. @JamieStumpORNL is this still a working feature?
> 
> > NOTE: To use multiple heat sources, use the * to denote the wildcard (ex: “Beam.*.txt”). The program will start at “Beam.1.txt” then “Beam.2.txt” and keep going until “Beam.x.txt” does not exist. The same goes for the paths. “Path.x.txt” would be used with the beam parameters found in “Beam.x.txt”

It should work. It hasn't been used in a while but will generate nonsense input for the meltpool geometry feature.

streeve commented:
> > Is the Visual Studio build no longer being supported? Fine if so, but a few things to note:

Not that it's not supported, it's just not any more special than any other build option. There was a lot of specific detail that I wouldn't support for any option. I added some general makefile info

> > * It might help people who are already using 3DThesis to make a note saying to refer to the documentation included with the Version 3.0 release for details.

We could backport this to 3.0.1 (and because of that separate new feature docs from docs for version 3.0)

> > * There are notes on compiler options that are recommended. Are these all captured in the CMake build and would there be use cases where someone might need to know to change the CMake build options?
> >
> > > To allow for parallel computing and vectorize calculations, some additional steps are needed (RECOMMENDED). All recommended changes can be found in the project properties (the big menu which contained the ability to add include directories). It is also recommended that the Platform is set to x64 and Configuration is changed to Active (unless debugging).
> > > 
> > > 1. Configuration Properties -> C/C++ -> Code Generation -> Floating Point Model = Fast
> > > 2. Configuration Properties -> C/C++ -> Language -> Conformance mode = No
> > > 3. Configuration Properties -> C/C++ -> Language -> Open MP Support = Yes
> 
> I guess not. It has been a while since I used VS but it should still build (before the mpi update)

From what I saw it was "msvc is faster", but yes you can pass any compiler. I'll add an example build in a follow on PR

> > On the multiple beam usage, this is the note in the Word document. @JamieStumpORNL is this still a working feature?
> > > NOTE: To use multiple heat sources, use the * to denote the wildcard (ex: “Beam.*.txt”). The program will start at “Beam.1.txt” then “Beam.2.txt” and keep going until “Beam.x.txt” does not exist. The same goes for the paths. “Path.x.txt” would be used with the beam parameters found in “Beam.x.txt”
> 
> It should work. It hasn't been used in a while but will generate nonsense input for the meltpool geometry feature.

Added that
